staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/metrics.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oidc
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"k8s.io/utils/clock"
+	"sync"
+	"time"
+
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	namespace = "apiserver"
+	subsystem = "authentication"
+)
+
+var (
+	jwtAuthenticatorLatencyMetric = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "jwt_authenticator_latency_seconds",
+			Help:           "Latency of jwt authentication operations in seconds. This is the time spent authenticating a token for cache miss only (i.e. when the token is not found in the cache).",
+			StabilityLevel: metrics.ALPHA,
+			// Start 0.1ms with the last bucket being [~52s, +Inf)
+			Buckets: metrics.ExponentialBuckets(0.0001, 2, 20),
```

Comment:
would like a sanity check from @logicalhan on the min / max / count of these buckets ... 

* can we change bucketing later without breaking consumers of this metric or do we have to get it perfect now?
* .1ms seems low as a minimum... do we actually care about distinctions between .1ms and 1ms? this is making 5 buckets between those. If we don't care, could just make the min 1ms.
* 20 steps seems like a lot... do we actually care about distinctions 20-30+ seconds and 50+ seconds? if not, could reduce the max to probably around 30 seconds

Reply:
1. You can change these later
2. .1ms does seem low as a min
3. Personally, I would probably hardcode the buckets I wanted

Reply:
DefaultBuckets have always seemed pretty reasonable to me for network request buckets. Ranges from 5ms to 10s. If we can change them later without impact, let's just use those (or maybe copy them and add a 1ms starting point)

Reply:
done!